### ISSUE TITLE
flutter build for iOS is dumber than for Android

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#8361]
## Steps to Reproduce

# Android

1. cd `examples/flutter_gallery`
1. wipe any caches pub files: `rm -rf .packages; rm pubspec.lock`
1. Build for android: `flutter build apk`

```
$ flutter build apk
Running 'flutter packages get' in flutter_gallery...  9.4s
Running 'gradle assembleRelease'...                   2.0s
Built app-release.apk (14.7MB).
```

=> flutter knows to do a `packages get`

# iOS

1. wipe any caches pub files: `rm -rf .packages; rm pubspec.lock`
1. Build for iOS: `flutter build ios`

```
$ flutter build ios
.packages does not exist.
Did you run "flutter packages get" in this directory?
```

=> flutter fails, and tells developer to manually get packages

Can we make them equally smart?

**Comment by mit-mit** [comment]
CC @kasperl issue you found

**Comment by mit-mit** [comment]
@chinmaygarde, @cbracken any idea why this is different on iOS?

**Comment by cbracken** [comment]
Unaware of any historical reason why it should be any different; sounds like a bug. Thanks for reporting!

**Comment by cbracken** [comment]
Fixed by #8412
